DuckDuckGo/SecureVaultErrorReporter.swift

```diff
@@ -27,34 +27,41 @@ final class SecureVaultErrorReporter: SecureVaultErrorReporting {
     static let shared = SecureVaultErrorReporter()
     private init() {}
 
+    @MainActor
```

Comment:
Moved the application state check here so it runs on `main` since I noticed while testing that it's sometimes being called from other threads 
![image](https://github.com/duckduckgo/iOS/assets/91189922/c98b5585-1cb5-4dff-92bd-ef3e3f8113d1)

Reply:
Nice, good find :+1: